sdk/keyvault/keyvault-secrets/src/index.ts

```diff
@@ -561,84 +466,10 @@ export class SecretClient {
     secretName: string,
     options: ListPropertiesOfSecretVersionsOptions = {},
   ): PagedAsyncIterableIterator<SecretProperties> {
-    const iter = this.listPropertiesOfSecretVersionsAll(secretName, options);
-
-    return {
-      next() {
-        return iter.next();
-      },
-      [Symbol.asyncIterator]() {
-        return this;
-      },
-      byPage: (settings: PageSettings = {}) =>
-        this.listPropertiesOfSecretVersionsPage(secretName, settings, options),
-    };
-  }
-
-  /**
-   * Deals with the pagination of {@link listPropertiesOfSecrets}.
-   * @param continuationState - An object that indicates the position of the paginated request.
-   * @param options - Optional parameters for the underlying HTTP request.
-   */
-  private async *listPropertiesOfSecretsPage(
-    continuationState: PageSettings,
-    options: ListPropertiesOfSecretsOptions = {},
-  ): AsyncIterableIterator<SecretProperties[]> {
-    if (continuationState.continuationToken == null) {
-      const optionsComplete: GetSecretsOptionalParams = {
-        maxresults: continuationState.maxPageSize,
-        ...options,
-      };
-      const currentSetResponse = await tracingClient.withSpan(
-        "SecretClient.listPropertiesOfSecretsPage",
-        optionsComplete,
-        (updatedOptions) => this.client.getSecrets(this.vaultUrl, updatedOptions),
-      );
-      continuationState.continuationToken = currentSetResponse.nextLink;
-      if (currentSetResponse.value) {
-        yield currentSetResponse.value.map(
-          (bundle: SecretBundle | DeletedSecretBundle) =>
-            getSecretFromSecretBundle(bundle).properties,
-        );
-      }
-    }
-    while (continuationState.continuationToken) {
-      const currentSetResponse = await tracingClient.withSpan(
-        "SecretClient.listPropertiesOfSecretsPage",
-        options,
-        (updatedOptions) =>
-          this.client.getSecretsNext(
-            this.vaultUrl,
-            continuationState.continuationToken!,
-            updatedOptions,
-          ),
-      );
-      continuationState.continuationToken = currentSetResponse.nextLink;
-      if (currentSetResponse.value) {
-        yield currentSetResponse.value.map(
-          (bundle: SecretBundle | DeletedSecretBundle) =>
-            getSecretFromSecretBundle(bundle).properties,
-        );
-      } else {
-        break;
-      }
-    }
-  }
-
-  /**
-   * Deals with the iteration of all the available results of {@link listPropertiesOfSecrets}.
-   * @param options - Optional parameters for the underlying HTTP request.
-   */
-  private async *listPropertiesOfSecretsAll(
-    options: ListPropertiesOfSecretsOptions = {},
-  ): AsyncIterableIterator<SecretProperties> {
-    const f = {};
-
-    for await (const page of this.listPropertiesOfSecretsPage(f, options)) {
-      for (const item of page) {
-        yield item;
-      }
-    }
+    return mapPagedAsyncIterable(
+      this.client.getSecretVersions(secretName, options),
```

Comment:
Random thought, do we know if `maxPageSize` works here? It's not defined on `PageOptions` in the generated code so am wondering if the property is now being ignored.

Cc @MaryGao

Reply:
`maxPageSize` is no longer support as a general option and we update the guideline also [here](https://github.com/Azure/azure-sdk/pull/7951/files).

@deyaaeldeen @xirzec I was wonderinf if we need to provide a standard way to expose this parameter for services still needing this option: https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-js+maxPageSize+language%3ATypeScript+samples-dev&type=code.

Reply:
In the mean time for Key Vault perhaps we can customize the paging helpers to accept maxPageSize and pass it as `maxresults` like Key Vault is expecting? I'm not sure what kind of effort would be needed for that

Reply:
@maorleger we discussed [this internally](https://github.com/Azure/autorest.typescript/issues/2802#issuecomment-2499459836) and I could work with for paginate helper to support maxPageSize if breaking is not acceptable.

Reply:
Thank you, I replied in the issue https://github.com/Azure/autorest.typescript/issues/2802#issuecomment-2504224437 and think we want to continue supporting maxPageSize in keyvault 🙏